src/XCCDF_POLICY/xccdf_policy_remediate.c

```diff
@@ -1286,7 +1403,240 @@ static int _xccdf_policy_generate_fix_other(struct oscap_list *rules_to_fix, str
+"# Modify size of partitions appropriately to reflect actual machine's hardware\n"
+"#\n"
+"# Remove Linux partitions from the system prior to creating new ones (optional)\n"
+"# --all	erase all partitions\n"
+"# --initlabel	initialize the disk label to the default based on the underlying architecture\n"
+"clearpart --all --initlabel\n"
+"reqpart\n"
+"\n"
+"# Create primary system partitions (required for installs)\n"
+"part /boot --fstype=xfs --size=512 --fsoptions=\"nodev,nosuid,noexec\"\n"
+"part pv.01 --grow --size=1\n"
+"\n"
+"# Create a Logical Volume Management (LVM) group (optional)\n"
+"volgroup VolGroup pv.01\n"
+"\n"
+"# Create particular logical volumes (optional)\n"
+"\nlogvol / --fstype=xfs --name=root --vgname=VolGroup --size=10240 --grow\n"
+);
+
+static int _generate_kickstart_logvol(struct kickstart_commands *cmds, int output_fd)
+{
+	_write_text_to_fd(output_fd, common_partition);
+	struct oscap_iterator *logvol_it = oscap_iterator_new(cmds->logvol);
+	while (oscap_iterator_has_more(logvol_it)) {
+		char *command = (char *) oscap_iterator_next(logvol_it);
+		_write_text_to_fd(output_fd, command);
+		_write_text_to_fd(output_fd, "\n");
+	}
+	oscap_iterator_free(logvol_it);
+	_write_text_to_fd(output_fd, "logvol swap --name=swap --vgname=VolGroup --size=2016\n");
+	return 0;
+}
+
+const char *common_kickstart_header = (
+"# Specify installation method to use for installation\n"
+"# To use a different one comment out the 'url' one below, update\n"
+"# the selected choice with proper options & un-comment it\n"
+"#\n"
+"# Install from an installation tree on a remote server via FTP or HTTP:\n"
+"# --url		the URL to install from\n"
+"#\n"
+"# Example:\n"
+"#\n"
+"# url --url=http://192.168.122.1/image\n"
+"#\n"
+"# Modify concrete URL in the above example appropriately to reflect the actual\n"
+"# environment machine is to be installed in\n"
+"#\n"
+"# Other possible / supported installation methods:\n"
+"# * install from the first CD-ROM/DVD drive on the system:\n"
+"#\n"
+"# cdrom\n"
+"#\n"
+"# * install from a directory of ISO images on a local drive:\n"
+"#\n"
+"# harddrive --partition=hdb2 --dir=/tmp/install-tree\n"
+"#\n"
+"# * install from provided NFS server:\n"
+"#\n"
+"# nfs --server=<hostname> --dir=<directory> [--opts=<nfs options>]\n"
+"#\n"
+"\n"
+"# Set language to use during installation and the default language to use on the installed system (required)\n"
+"lang en_US.UTF-8\n"
+"\n"
+"# Set system keyboard type / layout (required)\n"
+"keyboard --vckeymap us\n"
+"\n"
+"# Configure network information for target system and activate network devices in the installer environment (optional)\n"
+"# --onboot	enable device at a boot time\n"
+"# --bootproto	method to obtain networking configuration for device (default dhcp)\n"
+"network --onboot yes --bootproto dhcp\n"
+"\n"
+"# Set the system's root password (required)\n"
+"# Plaintext password is: server\n"
```

Comment:
I'd prefer not to put any passwords into generated KS files, but in this case maybe we could randomly generate the password every time (and write its readable form in the comment section) to avoid creating a potential security weakness?